src/universal/dnd/PrivateDragState.js

```diff
@@ -0,0 +1,30 @@
+import {columnArray} from 'universal/utils/constants';
+
+class PrivateDragState {
```

Comment:
so this is hacky.
React is great because it uses a vdom & 1-way data flow.
Unfortunately, in this instance we need each project to know where of the other projects are painted inside the client window.

So, we create an array of `components` which are nothing more than the `refs` to those components in that status column. We don't do anything with that array because we like to be lazy.

When a drag starts, we go through each of those components, look them up in the DOM tree, & find their vertical center position. we cache this array cuz touching the DOM is expensive.

Every ~14ms the `hover` callback is triggered while a component is being dragged. When that happens, we look at where the cursor is & see if the cursor is high or low enough to affect the sort order. This calculation is pretty cheap, but doing it every 14ms gets expensive. So, when we determine that the card hasn't moved enough to affect the sort order, we calculate the `minY` `maxY`. That way, we know nothing should happen unless we exceed that range. That makes it cheap.